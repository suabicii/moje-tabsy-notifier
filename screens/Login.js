import React, {useState} from "react";
import {View} from "react-native";
import {Text} from "react-native";
import UserInput from "../components/auth/UserInput";
import SubmitButton from "../components/auth/SubmitButton";

function Login() {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [loading, setLoading] = useState(false);

    const handleSubmit = () => {
        console.log('Button was pressed');
    }

    return (
        <View style={{flex: 1, justifyContent: "center"}}>
            <Text style={{
                color: "#414959",
                fontWeight: "400",
                fontSize: 32,
                marginBottom: 20,
                textAlign: "center"
            }}>
                Zaloguj się, aby otrzymywać powiadomienia
            </Text>
            <UserInput name="EMAIL" value={email} setValue={setEmail}/>
            <UserInput name="HASŁO" value={password} setValue={setPassword} secureTextEntry={true}/>
            <SubmitButton loading={loading} handleSubmit={handleSubmit}/>
        </View>
    );
}

export default Login;