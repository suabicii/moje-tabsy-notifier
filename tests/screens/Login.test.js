import React from "react";
import renderer from "react-test-renderer";
import Login from "../../screens/Login";

it('should correctly render Login screen', () => {
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
});