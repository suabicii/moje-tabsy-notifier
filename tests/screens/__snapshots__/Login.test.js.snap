// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Login screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#414959",
        "fontSize": 32,
        "fontWeight": "400",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Zaloguj się, aby otrzymywać powiadomienia
  </Text>
  <View
    style={
      Object {
        "marginHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#414959",
          "fontWeight": "300",
        }
      }
    >
      EMAIL
    </Text>
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={false}
      style={
        Object {
          "borderBottomColor": "#8e93a1",
          "borderBottomWidth": 0.5,
          "height": 48,
          "marginBottom": 30,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#414959",
          "fontWeight": "300",
        }
      }
    >
      HASŁO
    </Text>
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "#8e93a1",
          "borderBottomWidth": 0.5,
          "height": 48,
          "marginBottom": 30,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#78c2ad",
        "borderRadius": 24,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 15,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 24,
          "fontWeight": "300",
          "textAlign": "center",
        }
      }
    >
      Zaloguj się
    </Text>
  </View>
</View>
`;
